File: scala/sdktest/EmvDataEntityTest.scala
// Generated basic-flow test for the emv_data entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped EmvDataTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsmerchantservicessdk.core.{Helpers, BluefinTecsMerchantServicesSDK}
import voxgig.bluefintecsmerchantservicessdk.utility.struct.Struct

object EmvDataEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("emv_data.instance") {
      val testsdk = BluefinTecsMerchantServicesSDK.testSDK()
      val ent = testsdk.emvData(null)
      rep.check("emv_data.instance", ent != null, "expected non-null emv_data entity")
    }

    rep.scope("emv_data.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/emv_data/EmvDataTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsMerchantServicesSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("emv_data01", "EMV_DATA01")
      idmap.put("emv_data02", "EMV_DATA02")
      idmap.put("emv_data03", "EMV_DATA03")
      val now = System.currentTimeMillis()

      // CREATE
      val emvDataRef01Ent = client.emvData(null)
      var emvDataRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.emv_data"), "emv_data_ref01"))
      val emvDataRef01DataResult = emvDataRef01Ent.create(emvDataRef01Data, null)
      emvDataRef01Data = Helpers.toMapAny(emvDataRef01DataResult)
      rep.check("emv_data.create.map", emvDataRef01Data != null, "expected create result to be a map")
    }
  }
}

File: elixir/test/emv_data_entity_test.exs
# EmvData entity test (offline, mock transport)

defmodule BluefinTecsMerchantServices.EmvDataEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias BluefinTecsMerchantServices.Helpers, as: H
  alias BluefinTecsMerchantServices.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/emv_data/EmvDataTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    BluefinTecsMerchantServices.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.emv_data"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = BluefinTecsMerchantServices.test()
    ent = BluefinTecsMerchantServices.emv_data(sdk)
    assert ent != nil
  end

  test "should create then read back" do
    sdk = BluefinTecsMerchantServices.test(S.jm(["entity", S.jm(["emv_data", S.jm([])])]))
    ent = BluefinTecsMerchantServices.emv_data(sdk)
    made = BluefinTecsMerchantServices.Entity.EmvData.create(ent, S.jm(["name", "test-create"]))
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end
